src/components/Calculator.js

```diff
@@ -1,45 +1,61 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+
+    this.btnClicked = (e) => {
+      const clickedBtn = e.target.innerText;
+      const Result = calculate(this.state, clickedBtn);
+      this.setState(Result);
+    };
   }
 
   render() {
+    const { total, operation, next } = this.state;
     return (
       <div className="calculator">
         <div className="display">
-          <p>0</p>
+          <span>
+            {total}
+            {operation}
+            {next}
+          </span>
         </div>
         <div className="keys">
-          <button type="button">AC</button>
-          <button type="button">+/-</button>
-          <button type="button">%</button>
-          <button type="button" className="act">÷</button>
+          <button type="button" onClick={this.btnClicked}>AC</button>
+          <button type="button" onClick={this.btnClicked}>+/-</button>
+          <button type="button" onClick={this.btnClicked}>%</button>
+          <button type="button" className="act" onClick={this.btnClicked}>÷</button>
         </div>
         <div className="keys">
-          <button type="button">7</button>
-          <button type="button">8</button>
-          <button type="button">9</button>
-          <button type="button" className="act">X</button>
+          <button type="button" onClick={this.btnClicked}>7</button>
+          <button type="button" onClick={this.btnClicked}>8</button>
+          <button type="button" onClick={this.btnClicked}>9</button>
+          <button type="button" className="act" onClick={this.btnClicked}>X</button>
```

Comment:
- Great job for implementing most functionality :+1: however your multiplication operator is not working as expected because you used `X` and the condition is expecting `x` you can check that in your `src/logic/operate.js` line number 12  :+1:  please fix it to fit the requirement :+1: